modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx

```diff
@@ -635,7 +633,18 @@ function Filters({fdsFilterClientExtensions, fdsView, namespace}: IProps) {
 							if (newfilter.label === undefined) {
 								newfilter.label = '';
 							}
-							setFilters([...filters, newfilter]);
+
+							setFilters([newfilter, ...filters]);
+
+							let filterOrderString = newfilter.id.toString();
+
+							filters.map((filter) => {
+								filterOrderString += ',' + filter.id;
+							});
+
+							updateFDSFiltersOrder({
+								fdsFiltersOrder: filterOrderString,
+							});
```

Comment:
This is the quick & dirty solution. It is the worst of three solutions I can think of, as it involves two requests, one after another. Alternatives are:
1. Use `Promise.all()`. This way we still have two requests, but they are at least invoked in parallel.
2. Use a resource action, like we do with [saving fields](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/action/SaveFDSFieldsMVCResourceCommand.java). In this action backend, we update both fields and order, all in one request. This is probably most difficult to implement, but results in the best performance.

The forth way would be to somehow use headless API for this, like the `nestedFields` trick we do for [fetching multiple entities](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/filters/Filters.tsx#L473), but I don't think we can do something similar with POST endpoints. I am not sure, so this is worth confirming.

I don't think this bug has a high priority, so wen spend some time improving the solution. @dsanz is that right?

Reply:
I opened a discussion with you guy to talk about this. Priority is more a product decision. However, here are a few notes:
* If we update order persistence when creating, so shall we when deleting
* Action in backend might work if we make it generic enough, because we must account for filters, sorts, creation actions and item actions.